client/my-sites/domains/domain-management/list/index.jsx

```diff
@@ -344,7 +347,12 @@ export class List extends React.Component {
 	};
 
 	shouldUpgradeToMakeDomainPrimary( domain ) {
-		const { isDomainOnly, isOnFreePlan, hasNonPrimaryDomainsFlag } = this.props;
+		const {
+			isDomainOnly,
+			isOnFreePlan,
```

Comment:
Do we need to check this now? If they're on a paid plan then they should have the feature to set the primary domain.

Similarly would do we need to check for the 'domain only' plan still?

Reply:
100% agree, but I'd like to hold on removing it for now. I suggest creating a new follow-up PR.

Reply:
Ok let's do that. If we can, maybe we can add a test, so we can convince ourselves that the logic hasn't changed

Reply:
sounds very good to me.